File: java/src/main/java/co/edu/sena/capitulo6/ejercicios/Ejercicio04.java
package co.edu.sena.capitulo6.ejercicios;

import java.util.Scanner;

public class Ejercicio04 {

	public static void main(String[] args) {
		Scanner teclado = new Scanner(System.in);
		int numero, f1, f2, i, suma = 0;
		do {
			System.out.print("Introduce numero mayor que 1: ");
			numero = teclado.nextInt();
		} while (numero < 1);
		System.out.println("Los " + numero + " primeros términos de la serie de Fibonacci son:");

		f1 = 1;
		f2 = 1;

		System.out.print(f1 + " ");
		for (i = 2; i <= numero; i++) {
			System.out.print(f2 + " ");
			f2 = f1 + f2;
			f1 = f2 - f1;
		}
		suma = f1 + f2 - 1;
		System.out.println();
		System.out.println("La sumatoria es: " + suma);
	}

}

File: typescript/src/ejercicios/ejercicio04.ts
(function () {
    let numero,f1,f2,i:number;
    let suma = 0;
    do{
       numero=parseInt(prompt("Introduce numero mayor que 1: " ));
       f2=parseInt(prompt(f2 + " "));
       f1 = parseInt(prompt(f1 + " "));
    }while(numero<1);
    f1=1;
    f2=1; 
    for(i=2;i<=numero;i++){
         f2 = f1 + f2;
         f1 = f2 - f1;
    }
    suma = f1 + f2-1;
    console.log()
    console.log("La sumatoria es: " + suma);
    console.log("Los " + numero + " primeros términos de la serie de Fibonacci son:"); 

})();
